lib/Math/BaseCalc.pm

```diff
@@ -88,7 +88,12 @@ sub to_base {
     $num /= $dignum;
     #$num = (($num - ($num % $dignum))/$dignum);  # An alternative to the above
   }
-  return length $result ? $result : $self->{digits}[0];
+  my $resultlength = length $result;
+  if ($size && $size > $resultlength){
+    my $smallDigit  = $self->{digits}[0];
+    $result = sprintf("%${smallDigit}${size}s",$result);
```

Comment:
I think this will only work if `$self->{digits}[0]` is `"0"`, which we don't want to require.  Probably it just needs to prepend `$self->{digits}[0] x ($resultlength - $size)`.